### PR TITLE
EAGLE-283 Convert fid,uid in MapR's audit log to FIle/folder name, user name

### PR DESCRIPTION
**Pull request by DadanielZ**
1. Added resolver for user's input of "volume", "file/folder", "Status"
	- These resolver provide drop down list for user to select

2. Added NameConvertor to convert user's input to corresponding id and saved it in policy
	- volume value will be used to call MapR's REST API to get volume id
	- status code will be resolved to status id
	- file/folder name will be passed to MapR's command line class to get its id.
3. Modified controller.js
	- As soon as user input a value for "volume", "file", "status", it will fetch user's input and call rest service to get corresponding id
	- Save ids as a new match criteria in the policy

Eg: after user select "== '/test'" in field "dst", a new match criteria "== 234.567.890" will appear automatically under it, and the policy will end up as:
      dst == 'test'	or dst == '234.567.890'
	Same for volume and status input.
	In this way, the policy is human readable and can be used to match the ids in maprfsAuditLog for monitoring.

4. Added maprfs resource resolver to provide drop down list for user

5. Added license for "/eagle-app-example/src/main/java/org/apache/eagle/app/example/ExampleStormConfig.java" to pass rat check.

For user id conversion I haven't come up with a perfect solution, this is because:
- As user id is more OS standard staff, user might need to make OS calls to get the user ids.
- If user use LDAP, it will make it easier as we can query that directly.

### CODE REVIEW

**Comment by DadanielZ** [comment]
As I mentioned, user id conversion is a little bit tricky, it will be great if someone can give me a hint on user id conversion issue. 

Thanks and regards,
Da

**Comment by DadanielZ** [comment]
By the way, "id converter" will make http request to mapr host.
As mapr use https connection,  " javax.net.ssl.SSLHandshakeException Error" will occur if user didn't add the  certificate to JVM's trust store.

Will update this on related doc later.

Regards,
Da

File: eagle-security/eagle-security-maprfs-web/pom.xml [hunk 1]
```diff
@@ -64,6 +64,16 @@
...
       </dependency>
+      <dependency>
+          <groupId>com.mapr.hadoop</groupId>
+          <artifactId>maprfs</artifactId>
+          <version>${maprfs.version}</version>
+      </dependency>
+      <dependency>
+          <groupId>org.apache.httpcomponents</groupId>
+          <artifactId>httpclient</artifactId>
+          <version>4.5.2</version>
```
Comment (haoch) [inline eagle-security/eagle-security-maprfs-web/pom.xml]: Please avoid declaring any dependency with explicit version but use property defined in root pom instead. Better way is to define in dependency-management firstly and include without version.

File: eagle-security/eagle-security-maprfs-auditlog/src/main/resources/maprFSAuditLog-init.sh [hunk 2]
```diff
@@ -32,7 +32,7 @@ curl -u ${EAGLE_SERVICE_USER}:${EAGLE_SERVICE_PASSWD} -X POST -H 'Content-Type:a
            "application":"maprFSAuditLog"
         },
         "enabled": true,
-        "config": "classification.fs.defaultFS=maprfs:///"
+        "config": "classification.fs.defaultFS=hdfs://mapr1.da.dg:7222\nclassification.mapr.webhttps=https://mapr1.da.dg:8443\nclassification.mapr.username=mapr\nclassification.mapr.password=mapr"
```
Comment (haoch) [inline eagle-security/eagle-security-maprfs-auditlog/src/main/resources/maprFSAuditLog-init.sh]: Is `mapr1.da.dg` dataguise sensitive info? :-)

File: eagle-security/eagle-security-maprfs-web/src/main/test/java/MapRNameResolverTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import org.apache.eagle.service.security.hdfs.rest.MapRNameResolver;
+
+public class MapRNameResolverTest {
+
+    public static void main(String[] args){
```
Comment (haoch) [inline eagle-security/eagle-security-maprfs-web/src/main/test/java/MapRNameResolverTest.java]: Could you pls. refactor the `main` to a unit test case?

File: eagle-security/eagle-security-maprfs-web/src/main/test/java/org/apache/eagle/service/security/hdfs/rest/HttpRequestTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+package org.apache.eagle.service.security.hdfs.rest;
+
+import org.apache.eagle.service.security.hdfs.resolver.MAPRFSVolumeResolver;
+import org.json.JSONObject;
+
+import java.util.List;
+
+public class HttpRequestTest {
+    public static void main(String args[]) throws Exception {
+        String restUrl = "https://mapr1.da.dg:8443/rest/volume/list";
```
Comment (haoch) [inline eagle-security/eagle-security-maprfs-web/src/main/test/java/org/apache/eagle/service/security/hdfs/rest/HttpRequestTest.java]: - Could you pls. refactor the `main` to a unit test case ?
- Probably sensitive info?

File: eagle-webservice/src/main/resources/application.conf [hunk 5]
```diff
@@ -16,13 +16,24 @@
 {
 	eagle {
 		service {
-			storage-type = "hbase"
```
Comment (haoch) [inline eagle-webservice/src/main/resources/application.conf]: Please keep hbase storage as default store type

**Comment by haoch** [comment]
@DadanielZ reviewed as above.

File: eagle-webservice/src/main/resources/application.conf (hunk 5 above)
Comment (DadanielZ) [inline eagle-webservice/src/main/resources/application.conf]: thank you for the review，will update it and create a new PR

**Comment by haoch** [comment]
@DadanielZ thanks, but I think you don't have to reopen another new pull request but could continue to modify and commit on you current branch as any additional changes will append directly to this PR. 

But as a better practice, you are recommended to create an independent feature branch (typically named EAGLE-${JIRA_ID}) from latest apache `develop` branch in your own forked repository for every new pull request instead of using `develop` branch always :-)

**Comment by DadanielZ** [comment]
@haoch  thanks for your advice! XD 
Working on a feature branch will be a good practice.

**Pull request closed without merging**